src/users/users.service.ts

```diff
@@ -31,21 +31,42 @@ export class UsersService {
    */
   async createUser(userName: string, displayName: string): Promise<User> {
     const user = User.create(userName, displayName);
-    return this.userRepository.save(user);
+    try {
+      return await this.userRepository.save(user);
+    } catch {
+      this.logger.debug(
+        `A user with the username '${userName}' already exists.`,
+        'createUser',
+      );
+      throw new AlreadyInDBError(
+        `A user with the username '${userName}' already exists.`,
+      );
+    }
   }
 
   /**
    * @async
    * Delete the user with the specified userName
-   * @param userName - the username of the user to be delete
+   * @param {User} user - the username of the user to be delete
    * @throws {NotInDBError} the userName has no user associated with it.
    */
-  async deleteUser(userName: string): Promise<void> {
-    // TODO: Handle owned notes and edits
```

Comment:
Do we already handle this?

Reply:
The CASCADE deletes the notes. The revisions (which I presume is meant by edits here) are kept, because they aren't directly associated with the user.

Reply:
Only thing is, that revisions have authorships, which in turn refer to User, but they don't get deleted, when the user is deleted (I think this is the correct way of handling this), but we than don't get any user at all.

Reply:
To be clear revsions of your own notes get deleted, when the notes are deleted. But a user also has authorship of revisions in other notes…

Reply:
So the authorships stay in the DB, but get `NULL` set in their `user` foreign key?